purify/mail_style
# Almost complete rewrite for Rails 3 + bugfixes

Hi,

I've done an almost complete rewrite of the mail_style gem. Let me know what you think!

List of changes:
- Revamped test suite to be more contained, more descriptive and easier to work with
- Added support for autotest
- Separated code that handles mail and `ActionMailer` stuff from the code that inlines the styles, etc.
- Make inline `style` attribute already present before inlining have the final word
- Added support for `!important` styles
- Added support for `data` URIs in styles
- Make URLs in `style` attributes absolute
- Support URLs with parenthesis in them (`url(../images/image_(small).jpg)`)
- Don't add duplicate `<meta>` tags to `head`
- Cleaner code (IMO)

Let's have a discussion about this one. Some answers you probably want to have:
- I didn't enable support for both Rails 2 and Rails 3 for several reasons:
  - It already works in Rails 2 and has been a bit stale for a while ("not that active")
  - It would have made the changes take a longer time to make, and we kinda need this one at work
  - It would have increased complexity and made integration testing much harder to do
- I did some very opinionated changes to the code so I don't _expect_ them to be used
- I might make a formal fork if you don't like my approach
- I didn't add myself to the contributors section in the README since I'm not sure what to call me right now.
  Either I'm a normal contributor or I'm a co-author. I'm letting you decide this.
- I bumped up the version to `1.0.0.pre1` since I think this code is stable enough to be called a 1.0 release, and the API is incompatible and a version change from `0.1` to `1.0` should communicate this better than `0.1` to `0.2`.

There's nothing stopping us from adding Rails 2 support too, we just need to figure out how to test both at the same time in integration (or skip integration testing for Rails 2) and then create another ActionMailer mixin that targets ActionMailer 2. We also need users to manually require the correct file when loading the gem.

Thanks,
Magnus
